Firestore/core/include/firebase/firestore/listener_registration.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_INCLUDE_FIREBASE_FIRESTORE_LISTENER_REGISTRATION_H_
+#define FIRESTORE_CORE_INCLUDE_FIREBASE_FIRESTORE_LISTENER_REGISTRATION_H_
+
+namespace firebase {
+namespace firestore {
+
+class FirestoreInternal;
+class ListenerRegistrationInternal;
+
+/** Represents a listener that can be removed by calling remove. */
+class ListenerRegistration {
+ public:
+  /**
+   * @brief Default constructor. This creates a no-op instance.
+   */
+  ListenerRegistration();
+
+  /**
+   * @brief Copy constructor. It's totally okay to copy ListenerRegistration
+   * instances.
+   *
+   * @param[in] registration ListenerRegistration to copy from.
+   */
+  ListenerRegistration(const ListenerRegistration& registration);
+
+  /**
+   * @brief Move constructor. Moving is an efficient operation for
+   * ListenerRegistration instances.
+   *
+   * @param[in] registration ListenerRegistration to move data from.
```

Comment:
As above. (i.e. Personally, I'd remove this line too.)

Reply:
ack